app/src/main/java/org/torproject/android/OrbotMainActivity.java

```diff
@@ -817,105 +829,98 @@ private synchronized void updateStatus(String torServiceMsg, String newTorStatus
             mTxtOrbotLog.append(torServiceMsg + '\n');
         }
 
-        if (torStatus == null || (newTorStatus != null && newTorStatus.equals(torStatus))) {
-            torStatus = newTorStatus;
-            return;
-        } else {
-            torStatus = newTorStatus;
-        }
-
-        if (torStatus == null) {
-            return;
-        }
-
-        switch (torStatus) {
-            case TorServiceConstants.STATUS_ON:
+        if (!TextUtils.isEmpty(newTorStatus)) {
+            if (torStatus == null || (newTorStatus != null && newTorStatus.equals(torStatus))) {
```

Comment:
inconsequential, but `newTorStatus` should never be null there because `TextUtils.isEmpty` returned false